lucene/core/src/java/org/apache/lucene/codecs/blocktree/BlockTreeTermsWriter.java

```diff
@@ -1060,36 +1052,35 @@ public void close() throws IOException {
       return;
     }
     closed = true;
-    
+
+    final String metaName = IndexFileNames.segmentFileName(state.segmentInfo.name, state.segmentSuffix, BlockTreeTermsReader.TERMS_META_EXTENSION);
     boolean success = false;
-    try {
-      
-      final long dirStart = termsOut.getFilePointer();
-      final long indexDirStart = indexOut.getFilePointer();
+    try (IndexOutput metaOut = state.directory.createOutput(metaName, state.context)) {
+      CodecUtil.writeIndexHeader(metaOut, BlockTreeTermsReader.TERMS_META_CODEC_NAME, BlockTreeTermsReader.VERSION_CURRENT,
+          state.segmentInfo.getId(), state.segmentSuffix);
 
-      termsOut.writeVInt(fields.size());
+      metaOut.writeVInt(fields.size());
```

Comment:
@jpountz here I see the same lack of serializer write/read code, could it be possible to have such thing, It would improve readability and unit testing by only mocking fieldMetadatas and check serialization is correctly applied.

Reply:
Your proposal sounds orthogonal to this pull request to me?

Reply:
yes but maybe is the opportunity to move the code and add more testability in BlockTreeTermsReader.java

Reply:
I prefer decoupling changes when possible.